### PR TITLE
Support bookinfo ingress gw on OCP in install-testing-demos script

### PR DESCRIPTION
**Pull request by mkralik3**
- Support bookinfo ingress gw on OCP in install-testing-demos script (moved gateway_yaml before if-else and used it for openshift case too )
- Add route for bookinfo ingress gateway in bookinfo ns
- Use that route in traffic generator when exists (1st, it tries `istio-ingressgateway` route (in ingressgateway/controlplane ns), 2nd it tries bookinfo-gateway-istio (in bookinfo app ns), 3rd it uses a direct route to productpage

### CODE REVIEW

File: hack/istio/install-bookinfo-demo.sh [hunk 1]
```diff
@@ -399,12 +400,21 @@ fi
...
   if [ "${IS_OPENSHIFT}" == "true" ]; then
-    echo "Determining the route to send traffic to"
+    echo "Determining the route to send traffic to, 1st, trying istio-ingressgateway route in ${INGRESS_NAMESPACE} namespace"
+    # first, try istio-ingressgateway in istio-system
     INGRESS_ROUTE=$(${CLIENT_EXE} get route istio-ingressgateway -o jsonpath='{.spec.host}{"\n"}' -n ${INGRESS_NAMESPACE})
-    while [ -z "${INGRESS_ROUTE}" ]; do
+    if [ -z "${INGRESS_ROUTE}" ]; then
       sleep 1
-      INGRESS_ROUTE=$(${CLIENT_EXE} get route productpage -o jsonpath='{.spec.host}{"\n"}' -n ${NAMESPACE})
-    done
```
Comment (nrfox) [inline hack/istio/install-bookinfo-demo.sh:407]: I'm guessing this was a while loop before to give the ingress route time to be populated? We may need to check more than once before giving up and going onto bookinfo-gateway-istio. Maybe wait for the ingress status to have a `host` for some reasonable timeout?
```
kubectl wait '--for=jsonpath={.status.ingress[].host}' --timeout=10s route istio-ingressgateway -n ${INGRESS_NAMESPACE}
```

Comment (mkralik3) [reply hack/istio/install-bookinfo-demo.sh:407]: ah, yea. 
However, to support JSONPath condition without value, some newer kubectl/oc is needed, I am not sure which version is used in our CI

Comment (nrfox) [reply hack/istio/install-bookinfo-demo.sh:407]: Ok maybe then just keep the while loop but exit after ~10 tries?

**Review by nrfox — APPROVED** [review]

File: hack/istio/install-bookinfo-demo.sh (hunk 1 above)
Comment (mkralik3) [reply hack/istio/install-bookinfo-demo.sh:407]: It works on our CI with
```
oc version
Client Version: 4.16.6
Kustomize Version: v5.0.4-0.20230601165947-6ce0bf390ce3
Server Version: 4.17.4
Kubernetes Version: v1.30.5
```
so I left it in that way.

**Pull request merged**